storage/innobase/trx/trx0purge.cc

```diff
@@ -1189,7 +1189,10 @@ dict_table_t *purge_sys_t::close_and_reopen(table_id_t id, THD *thd,
 
   dict_table_t *table= trx_purge_table_open(id, mdl_context, mdl);
   if (table == reinterpret_cast<dict_table_t*>(-1))
+  {
+    VALGRIND_YIELD;
     goto retry;
+  }
```

Comment:
We could avoid some code duplication by introducing another label here and replacing the subsequent `goto retry` jump here. But then again, this only affects Valgrind builds.